sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,347 @@
+                                                        child.frame.y)
+    assert cylindrical_joint.parent_axis == parent.frame.x
+    assert cylindrical_joint.child_axis == child.frame.x
+
+    x = dynamicsymbols('cylindrical_joint_x')
+    xd = dynamicsymbols('cylindrical_joint_x', 1)
+    v = dynamicsymbols('cylindrical_joint_v')
+
+    theta = dynamicsymbols('cylindrical_joint_theta')
+    thetad = dynamicsymbols('cylindrical_joint_theta', 1)
+    omega = dynamicsymbols('cylindrical_joint_omega')
+
+    assert cylindrical_joint.coordinates == [x, theta]
+    assert cylindrical_joint.speeds == [v, omega]
+    assert cylindrical_joint.kds == [xd - v, thetad - omega]
+
+    child_joint_point = cylindrical_joint.child_joint_point
+    parent_joint_point = cylindrical_joint.parent_joint_point
+    assert parent_joint_point.name == 'cylindrical_joint_parent_joint'
+    assert child_joint_point.name == 'cylindrical_joint_child_joint'
+    assert child_joint_point.pos_from(parent_joint_point) == x * parent.frame.x
+    child_joint_masscenter = l * (child.frame.x + child.frame.y)
+    assert child_joint_point.pos_from(child.masscenter) == child_joint_masscenter
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
+    assert child.frame.ang_vel_in(parent.frame) == omega * parent.frame.x
+    assert child.masscenter.vel(parent.frame) == - l * omega * child.frame.z
+    assert child_joint_point.vel(parent.frame) == v * parent.frame.x
+    assert parent.frame.dcm(child.frame) == Matrix([
+        [1,          0,           0],
+        [0, cos(theta), -sin(theta)],
+        [0, sin(theta),  cos(theta)]])
+
+
+def test_cylindrical_joint_arguments():
+    parent = Body('parent')
+    child = Body('child')
+    l1 = Symbol('l1')
+    l2 = Symbol('l2')
+    cylindrical_joint = CylindricalJoint('cylindrical_joint', parent, child,
+                                         parent_point_pos=(0, l1, 0),
+                                         child_point_pos=(0, l2, 0),
+                                         parent_axis='y', child_axis='z')
+    assert cylindrical_joint.name == 'cylindrical_joint'
+    assert cylindrical_joint.parent == parent
+    assert cylindrical_joint.child == child
+    assert cylindrical_joint.parent_axis == parent.frame.y
+    assert cylindrical_joint.child_axis == child.frame.z
+    assert cylindrical_joint._parent_joint_location == l1 * parent.frame.y
+    assert cylindrical_joint._child_joint_location == l2 * child.frame.y
+
+
+def test_planar_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    planar_joint = PlanarJoint('planar_joint', parent, child, child_axis='y',
+                               parent_distance=l)
+
+    assert planar_joint.name == 'planar_joint'
+    assert planar_joint.parent == parent
+    assert planar_joint.child == child
+    assert planar_joint._parent_joint_location == l * parent.frame.z
```

Comment:
Why is the default axis 'z'? I would think you'd make the default 'x' for everything.